contracts/Marketplace.sol

```diff
@@ -149,6 +156,8 @@ contract Marketplace is SlotReservations, Proofs, StateRetrieval, Endian {
     TokensPerSecond pricePerSecond = request.ask.pricePerSecond();
     uint128 price = pricePerSecond.accumulate(request.ask.duration);
 
+    _collectProtocolFee(request.client, request.ask);
```

Comment:
I'm assuming this function and `_transferToVault(request.client, fund, account, price);` would revert with "insufficient funds" if the user does not have enough funds to cover both the fee and the price?